src/controllers/csv-processor.ts

```diff
@@ -161,55 +179,30 @@ export const moveCSVFromBlobStorageToDatalake = async (dataset: Dataset): Promis
     }
 };
 
-export const uploadCSVToBlobStorage = async (buff: Buffer, dataset: Dataset): Promise<UploadDTO | UploadErrDTO> => {
+export const uploadCSVToBlobStorage = async (fileStream: Readable, filetype: string): Promise<Import> => {
     const blobStorageService = new BlobStorageService();
-    if (buff) {
-        const hash = createHash('sha256').update(buff).digest('hex');
-        const datafile = Datafile.createDatafile(dataset, hash.toString(), 'BetaUser');
-        const savedDataFile = await datafile.save();
-        const dto = await datasetToDatasetDTO(dataset);
+    if (fileStream) {
+        const importRecord = new Import();
+        importRecord.id = randomUUID();
+        importRecord.mime_type = filetype;
         try {
-            logger.debug(`Uploading file ${savedDataFile.id} to blob storage`);
-            await blobStorageService.uploadFile(`${savedDataFile.id}.csv`, buff);
-            return {
-                success: true,
-                dataset: dto
-            };
+            await blobStorageService.uploadFile(`${importRecord.id}.csv`, fileStream);
```

Comment:
We need certainty the operation has completed successfully.  We can't fire and forget because if something goes wrong we could end up loosing the upload and having no data in the cube despite everything that the database tells us.

Reply:
We don't need to force everything to be synchronous tho': we only need to know at the point we commit the other objects that rely on it. For example, the DB objects we stash its name in.